packages/xrpl/src/Wallet/fundWallet.ts

```diff
@@ -20,15 +20,103 @@ const INTERVAL_SECONDS = 1
+ * xrpAmount parameter is set to '10', which is equivalent to 10 XRP.
+ *
+ * If the transaction is successful, the function logs a message to the console with the wallet public key and balance of the
+ * new wallet. If there's an error funding XRP to the wallet, the function logs an error message to the console using
+ * console.error().
+ *
+ * ```ts
  * const api = new xrpl.Client("wss://s.altnet.rippletest.net:51233")
  * await api.connect()
  * const { wallet, balance } = await api.fundWallet()
  * ```
  *
+ * Example 2: Fund wallet using a custom faucet host and known wallet address
+ *
+ * It's important to note that you can customize the faucetHost and faucetPath options to
+ * specify the endpoint that the fundWallet() method uses to send XRP to a wallet. The faucetHost option
+ * specifies the hostname of the server to connect to, and the faucetPath option specifies the path to the
+ * endpoint on the server. By default, the fundWallet() method uses the XRP Ledger Testnet
+ * faucet at https://faucet.altnet.rippletest.net/accounts.
+ *
+ * ```ts
+ * const walletAddress = 'rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh'
+ *
+ * async function fundWalletWithCustomfaucetHost(address) {
+ *   try {
+ *     const { balance } = await client.fundWallet(address, {
+ *       xrpAmount: '10',
+ *       faucetHost: 'https://custom-faucet.example.com',
+ *       faucetPath: '/accounts'
+ *     })
+ *     console.log(`Sent 10 XRP to wallet: ${address}, balance: ${balance} XRP`)
+ *   } catch (error) {
+ *     console.error(`Failed to fund wallet: ${error}`)
+ *   }
+ * }
+ *
+ * fundWalletWithCustomfaucetHost(walletAddress)
+ * ```
+ *
+ * Example 3: Call `fundWallet` with no parameters to generate a new wallet and fund it with testnet
+ *
+ * The easiest way to use the `fundWallet` method is with no parameters in order to generate a new wallet
+ * and fund it with the XRP Ledger Testnet:
+ *
+ * ```ts
+ * const { Client } = require('xrpl')
+ * const client = new Client('wss://s.altnet.rippletest.net:51233')
+ *
+ * async function fundNewWallet() {
+ *   try {
+ *     const { wallet, balance } = await client.fundWallet()
+ *     console.log(`Sent 10 XRP to wallet: ${wallet.publicKey}, balance: ${balance} XRP`)
+ *   } catch (error) {
+ *     console.error(`Failed to fund wallet: ${error}`)
+ *   }
+ * }
+ *
+ * fundNewWallet()
+ * ```
+ *
+ * In this example, we call the `fundWallet()` method with no parameters to generate a new
+ * wallet and fund it with the XRP Ledger Testnet. The wallet variable contains an object that has publicKey
+ * and secret properties for the new wallet, and the balance variable contains the balance of the new wallet.
+ *
+ * If the transaction is successful, the function logs a message to the console with the publicKey value of the
+ * new wallet and the balance of the wallet. If there's an error funding XRP to the wallet, the function logs an
+ * error message to the console using console.error().
+ *
```

Comment:
Recommended moving this example to the top so they can find the happy path first, instead of starting with all the optional parameters.
```suggestion
```